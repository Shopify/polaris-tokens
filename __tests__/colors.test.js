/* eslint-env jest */

const fs = require('fs');
const path = require('path');
const _ = require('lodash');

const colorsJSON = require('../dist/colors.json');
const colorsAseJSON = require('../dist/colors.ase.json');
const colorsRawJSON = require('../dist/colors.raw.json');

const colorFiles = [
  'colors.android-colors.xml',
  'colors.color-map.scss',
  'colors.custom-properties.css',
  'colors.scss',
  'Polaris.sketchpalette',
  'index.d.ts',
];

colorFiles.map((filename) =>
  it(`renders ${filename} correctly`, () => {
    const contents = fs.readFileSync(
      path.join(__dirname, '..', 'dist', filename),
      {
        encoding: 'utf-8',
      },
    );
    expect(contents).toMatchSnapshot();
  }),
);

describe('JSON objects', () => {
  it('renders similar JSON objects', () => {
    expect(
      _(colorsJSON)
        .toPairs()
        .sortBy(0)
        .fromPairs()
        .value(),
    ).toMatchSnapshot();
  });
  it('renders similar ASE JSON objects', () => {
    expect(colorsAseJSON.version).toBe('1.0');
    expect(colorsAseJSON.groups.length).toBe(0);
    expect(_(colorsAseJSON.colors).sortBy('name')).toMatchSnapshot();
  });
  it('renders similar RAW JSON objects', () => {
    // Aliases
    expect(
      _(colorsRawJSON.aliases)
        .toPairs()
        .sortBy(0)
        .fromPairs()
        .value(),
    ).toMatchSnapshot();

    // Props
    expect(_(colorsRawJSON.props).sortBy('name')).toMatchSnapshot();
  });
});
