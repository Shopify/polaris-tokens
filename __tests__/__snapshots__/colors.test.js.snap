// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON object representation renders similar ASE JSON objects 1`] = `
Array [
  Object {
    "color": Array [
      0.3137254901960784,
      0.1411764705882353,
      0.5607843137254902,
    ],
    "model": "RGB",
    "name": "dark",
    "type": "global",
  },
  Object {
    "color": Array [
      0.13725490196078433,
      0,
      0.3176470588235294,
    ],
    "model": "RGB",
    "name": "darker",
    "type": "global",
  },
  Object {
    "color": Array [
      0.8901960784313725,
      0.8156862745098039,
      1,
    ],
    "model": "RGB",
    "name": "light",
    "type": "global",
  },
  Object {
    "color": Array [
      0.9647058823529412,
      0.9411764705882353,
      0.9921568627450981,
    ],
    "model": "RGB",
    "name": "lighter",
    "type": "global",
  },
  Object {
    "color": Array [
      0.611764705882353,
      0.41568627450980394,
      0.8705882352941177,
    ],
    "model": "RGB",
    "name": "purple",
    "type": "global",
  },
  Object {
    "color": Array [
      0.3137254901960784,
      0.28627450980392155,
      0.35294117647058826,
    ],
    "model": "RGB",
    "name": "text",
    "type": "global",
  },
]
`;

exports[`JSON object representation renders similar JSON objects 1`] = `
Object {
  "color-dark": "rgb(80, 36, 143)",
  "color-darker": "rgb(35, 0, 81)",
  "color-light": "rgb(227, 208, 255)",
  "color-lighter": "rgb(246, 240, 253)",
  "color-purple": "rgb(156, 106, 222)",
  "color-text": "rgb(80, 73, 90)",
}
`;

exports[`JSON object representation renders similar RAW JSON objects 1`] = `
Object {
  "color-dark": Object {
    "value": "#50248f",
  },
  "color-darker": Object {
    "value": "#230051",
  },
  "color-light": Object {
    "value": "#e3d0ff",
  },
  "color-lighter": Object {
    "value": "#f6f0fd",
  },
  "color-purple": Object {
    "value": "#9c6ade",
  },
  "color-text": Object {
    "value": "#50495a",
  },
}
`;

exports[`JSON object representation renders similar RAW JSON objects 2`] = `
Array [
  Object {
    "category": "background-color",
    "name": "color-dark",
    "originalValue": "{!color-dark}",
    "type": "color",
    "value": "rgb(80, 36, 143)",
  },
  Object {
    "category": "background-color",
    "name": "color-darker",
    "originalValue": "{!color-darker}",
    "type": "color",
    "value": "rgb(35, 0, 81)",
  },
  Object {
    "category": "background-color",
    "name": "color-light",
    "originalValue": "{!color-light}",
    "type": "color",
    "value": "rgb(227, 208, 255)",
  },
  Object {
    "category": "background-color",
    "name": "color-lighter",
    "originalValue": "{!color-lighter}",
    "type": "color",
    "value": "rgb(246, 240, 253)",
  },
  Object {
    "category": "background-color",
    "name": "color-purple",
    "originalValue": "{!color-purple}",
    "type": "color",
    "value": "rgb(156, 106, 222)",
  },
  Object {
    "category": "text-color",
    "name": "color-text",
    "originalValue": "{!color-text}",
    "type": "color",
    "value": "rgb(80, 73, 90)",
  },
]
`;

exports[`renders Polaris.sketchpalette correctly 1`] = `
"{
  \\"compatibleVersion\\": \\"2.0\\",
  \\"pluginVersion\\": \\"2.0\\",
  \\"colors\\": [
    {
      \\"red\\": 0.3137254901960784,
      \\"green\\": 0.28627450980392155,
      \\"blue\\": 0.35294117647058826,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {
      \\"red\\": 0.13725490196078433,
      \\"green\\": 0,
      \\"blue\\": 0.3176470588235294,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {
      \\"red\\": 0.3137254901960784,
      \\"green\\": 0.1411764705882353,
      \\"blue\\": 0.5607843137254902,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {
      \\"red\\": 0.611764705882353,
      \\"green\\": 0.41568627450980394,
      \\"blue\\": 0.8705882352941177,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {
      \\"red\\": 0.8901960784313725,
      \\"green\\": 0.8156862745098039,
      \\"blue\\": 1,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {
      \\"red\\": 0.9647058823529412,
      \\"green\\": 0.9411764705882353,
      \\"blue\\": 0.9921568627450981,
      \\"alpha\\": 1
    },
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1},
    {\\"red\\": 1, \\"green\\": 1, \\"blue\\": 1, \\"alpha\\": 1}
  ],
  \\"gradients\\": [],
  \\"images\\": []
}
"
`;

exports[`renders colors.android.xml correctly 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<resources>
    <color name=\\"polaris_color_dark\\">#ff50248f</color>
    <color name=\\"polaris_color_darker\\">#ff230051</color>
    <color name=\\"polaris_color_light\\">#ffe3d0ff</color>
    <color name=\\"polaris_color_lighter\\">#fff6f0fd</color>
    <color name=\\"polaris_color_purple\\">#ff9c6ade</color>
    <color name=\\"polaris_color_text\\">#ff50495a</color>
</resources>"
`;

exports[`renders colors.color-map.scss correctly 1`] = `
"$polaris-colors: (
  'lighter': (
    'base': rgb(246, 240, 253),
  ),
  'light': (
    'base': rgb(227, 208, 255),
  ),
  'purple': (
    'base': rgb(156, 106, 222),
  ),
  'dark': (
    'base': rgb(80, 36, 143),
  ),
  'darker': (
    'base': rgb(35, 0, 81),
  ),
  'text': (
    'base': rgb(80, 73, 90),
  ),
);
"
`;

exports[`renders colors.custom-properties.css correctly 1`] = `
":root {
  --color-lighter: rgb(246, 240, 253);
  --color-light: rgb(227, 208, 255);
  --color-purple: rgb(156, 106, 222);
  --color-dark: rgb(80, 36, 143);
  --color-darker: rgb(35, 0, 81);
  --color-text: rgb(80, 73, 90);
}
"
`;

exports[`renders colors.scss correctly 1`] = `
"$color-lighter: rgb(246, 240, 253);
$color-light: rgb(227, 208, 255);
$color-purple: rgb(156, 106, 222);
$color-dark: rgb(80, 36, 143);
$color-darker: rgb(35, 0, 81);
$color-text: rgb(80, 73, 90);
"
`;

exports[`renders index.d.ts correctly 1`] = `
"declare interface Tokens {
  colorDark: \\"rgb(80, 36, 143)\\";
  colorDarker: \\"rgb(35, 0, 81)\\";
  colorLight: \\"rgb(227, 208, 255)\\";
  colorLighter: \\"rgb(246, 240, 253)\\";
  colorPurple: \\"rgb(156, 106, 222)\\";
  colorText: \\"rgb(80, 73, 90)\\";
  fontStackBase: \\"-apple-system, 'BlinkMacSystemFont', 'San Francisco', 'Roboto', 'Segoe UI', 'Helvetica Neue', sans-serif\\";
  fontStackMonospace: \\"Monaco, Consolas, 'Lucida Console', monospace\\";
  spacingBase: \\"16px\\";
  spacingExtraLoose: \\"32px\\";
  spacingExtraTight: \\"4px\\";
  spacingLoose: \\"20px\\";
  spacingNone: 0;
  spacingTight: \\"8px\\";
}
declare let tokens: Tokens;
export = tokens;"
`;
